app.py

```diff
@@ -2,41 +2,84 @@
 from os import path
 if path.exists("env.py"):
     import env
-from flask import Flask, render_template, url_for, flash, redirect
+from flask import Flask, render_template, url_for, flash, redirect, session
 from forms import RegistrationForm, LoginForm
 from flask_pymongo import PyMongo
 from bson.objectid import ObjectId
-
+from flask_login import LoginManager, UserMixin, login_user, login_required, logout_user, current_user
+from werkzeug.security import generate_password_hash, check_password_hash
+from flask_bcrypt import Bcrypt
 
 
 app = Flask(__name__)
 app.config["SECRET_KEY"] = os.environ.get('SECRET_KEY')
 app.config["MONGO_DBNAME"] = os.environ.get('MONGO_DBNAME')
 app.config["MONGO_URI"] = os.environ.get('MONGO_URI')
 
+
 mongo = PyMongo(app)
+bcrypt = Bcrypt(app)
+
+
+"""
+USER MANAGEMENT
+"""
+
+# Login
 
 @app.route('/login', methods=['GET', 'POST'])
 def login():
     form = LoginForm()
+    users = mongo.db.users
     if form.validate_on_submit():
-        if form.username.data == 'simon' and form.password.data == 'password':
-            flash('You are logged in', 'success')
+        login_user = users.find_one({'username' : form.username.data})
+        if login_user and bcrypt.check_password_hash(login_user['password'], form.password.data.encode('utf-8')):
+            session["username"] = form.username.data
+            flash(f'Welcome to squirrel, {form.username.data}.', 'success')
             return redirect(url_for('listing'))
         else:
-            flash('Wrong username or password', 'danger')
+            flash('Wrong username or password.', 'danger')
 
     return render_template('pages/login.html', title="Login", form=form)
 
 
+# Registration
+
 @app.route('/register', methods=['GET', 'POST'])
 def register():
     form = RegistrationForm()
+    users = mongo.db.users
     if form.validate_on_submit():
-        flash(f'Account created for {form.username.data}.', 'success')
-        return redirect(url_for('listing'))
+        existing_user = users.find_one({'username' : form.username.data})
+        existing_email = users.find_one({'email': form.email.data})
+        if existing_user is None and existing_email is None:
+            hashed_password = bcrypt.generate_password_hash(form.password.data).decode('utf-8')
+            users.insert({
+                "username": form.username.data,
+                "email": form.email.data,
+                "password": hashed_password,
+            })
+            flash(f'Account created for {form.username.data}.', 'success')
+            return redirect(url_for('login'))
+        elif existing_user:
+            flash(f'Username {form.username.data} is already in use.', 'danger')
```

Comment:
In my opinion, usernames should not be unique in your service: emails are already filling that purpose. If multiple people named Charlie want to use your service and be named Charlie within it, it’s a shame they can’t because someone else was there first. I would suggest making the username purely cosmetic and not enforcing uniqueness.